spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/TestSparkWriteConfDistributionMode.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark;
+
+import static org.apache.iceberg.TableProperties.WRITE_DISTRIBUTION_MODE;
+import static org.apache.iceberg.TableProperties.WRITE_DISTRIBUTION_MODE_HASH;
+import static org.apache.iceberg.TableProperties.WRITE_DISTRIBUTION_MODE_NONE;
+import static org.apache.iceberg.TableProperties.WRITE_DISTRIBUTION_MODE_RANGE;
+
+import java.util.Map;
+import org.apache.iceberg.DistributionMode;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestSparkWriteConfDistributionMode extends SparkTestBaseWithCatalog {
```

Comment:
Any reason we need a separate test class and not just add to `TestSparkDistributionAndOrderingUtil` ?

Reply:
yes, good call. Initially I was thinking about putting the logic into `TestSparkDistributionAndOrderingUtil` but then I realized the change I made is only about influencing the precedence level in `SparkWriteConf`. 
My change can be individually tested without any call to use method in `SparkDistributionAndOrderingUtil` class, so it does not seem make sense to add more complexity to 2000+ line of unit tests. 

Looks like I dont find any existing tests focus on the SparkWriteConf alone, I just created a this new one and it seems neat.

Reply:
What about calling it `TestSparkWriteConf` so that it becomes generic? We may add more tests for other configs in the future. I think it is fine to purely test the config.